code/game/objects/items/weapons/weaponry.dm

```diff
@@ -467,3 +467,73 @@ obj/item/weapon/banhammer/admin
 	sharpness = 2
 	sharpness = SHARP_TIP
 	hitsound = 'sound/weapons/bladeslice.ogg'
+
+/obj/item/weapon/baseball_bat
+	name = "baseball bat"
+	desc = "Good for reducing a doubleheader to a no-header."
+	hitsound = "sound/weapons/baseball_hit_flesh.ogg"
+	icon_state = "baseball_bat"
+	item_state = "baseball_bat"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/swords_axes.dmi', "right_hand" = 'icons/mob/in-hand/right/swords_axes.dmi')
+	flags = TWOHANDABLE
+	force = 15
+	throwforce = 10
+	throw_speed = 1
+	throw_range = 7
+	w_class = W_CLASS_LARGE
+
+/obj/item/weapon/baseball_bat/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(istype(W, /obj/item/stack/rods))
+		if(!(istype(user.get_inactive_hand(), /obj/item/weapon/storage/toolbox) || istype(user.get_inactive_hand(), /obj/item/weapon/hammer)))
+			to_chat(user, "<span class='info'>You need to be holding a toolbox or hammer to do that!</span>")
+			return
+		to_chat(user, "<span class='notice'>You hammer the [W.name] into \the [src].</span>")
+		var/obj/item/stack/rodstack = W
+		rodstack.use(1)
+		new /obj/item/weapon/spiked_bat(get_turf(src))
+		qdel(src)
+
+/obj/item/weapon/baseball_bat/IsShield()
+	return TRUE
+
+/obj/item/weapon/baseball_bat/on_block(damage, atom/movable/blocked)
+	if(ismob(loc))
+		var/mob/H = loc
+		if(!H.in_throw_mode || !wielded || damage > 15)
```

Comment:
given that it MATTERS that you wield this, this really needs to have separate inhands for wielded vs unwielded

Reply:
Working on this now.

Reply:
Asked Skowron to make sprites for this and he said he would, just waiting on that otherwise the PR is ready to go.